staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -178,7 +181,9 @@ func (cnc *CloudNodeController) Run(stopCh <-chan struct{}, controllerManagerMet
 		}
 	}, cnc.nodeStatusUpdateFrequency, stopCh)
 
-	go wait.Until(cnc.runWorker, time.Second, stopCh)
+	for i := int32(0); i < cnc.workerCount; i++ {
+		go wait.Until(cnc.runWorker, time.Second, stopCh)
+	}
 
 	<-stopCh
```

Comment:
Would it make sense to use a waitGroup and ensure that all the worker threads are actually stopping?

Reply:
I don't think its necessary but to be honest I'am not 100% sure.
I've based this on https://github.com/kubernetes/kubernetes/blob/759785ea147bc13945d521eaba4a6592cbc0675f/pkg/controller/nodeipam/ipam/range_allocator.go#L173 and https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/sample-controller/controller.go#L167

Reply:
should be ok since the worker go routines are stopped by the workqueue shutting down.